src/monodroid/jni/monodroid-glue.cc

```diff
@@ -2635,6 +2641,27 @@ MonodroidRuntime::get_java_class_name_for_TypeManager (jclass klass)
 	return ret;
 }
 
+void
+MonodroidRuntime::monodroid_javasystem_loadLibrary (const char *libname) noexcept
+{
+	JNIEnv *env = osBridge.ensure_jnienv ();
+	jmethodID loadLibraryID = env->GetStaticMethodID (mono_android_Runtime, "loadLibrary", "(Ljava/lang/String;)Z");
+	if (loadLibraryID == nullptr) {
+		log_error (LOG_DEFAULT, "Failed to look up the mono.android.Runtime.loadLibrary Java method");
+		return;
+	}
+
+	jstring libname_java = env->NewStringUTF (libname);
+	jboolean success = env->CallStaticBooleanMethod (mono_android_Runtime, loadLibraryID, libname_java);
+	if (success != 0) {
+		// Loaded
+		return;
+	}
+
+	log_warn (LOG_DEFAULT, "System.loadLibrary failed, attempting to load the library directly");
```

Comment:
Might be handy if we printed out *what* library failed to load, in case that's needed for future debugging.